sqlglot/expressions.py

```diff
@@ -219,10 +220,21 @@ def type(self, dtype: t.Optional[DataType | DataType.Type | str]) -> None:
             dtype = DataType.build(dtype)
         self._type = dtype  # type: ignore
 
+    @property
+    def metadata(self) -> t.Optional[t.Dict[str, t.Any]]:
+        return self._meta
+
+    def set_metadata(self, meta_key, meta_value):
```

Comment:
i don't really like this method, let's remove it, maybe you can just do this in the getter

@property
def meta(self) -> t.Dict[str, t.Any]:
   if self._meta is None:
       self._meta = {}
  return self._meta

then it's always available and you don't need this at all

Reply:
There was some discussion to lazily initialise this dictionary as it could be an empty dictionary majority of the time. It would only be populated when adding metadata

Reply:
it's similar in that it's only going to get set if you ask for metadata